File: cpp/modulino.cpp
// Copyright (C) YelloSoft

#include "./modulino.hh"

#include <iostream>

int meaning_of_life() {
    return 42;
}

#ifdef MODULINO

int main() {
    std::cout << "Main: The meaning of life is " << meaning_of_life() << std::endl;
    return 0;
}

#endif

File: pascal/modulino.pas
{$IFDEF modulino}
program Modulino;
{$ELSE}
unit Modulino;
interface
function MeaningOfLife () : integer;
implementation
{$ENDIF}
function MeaningOfLife () : integer;
begin
    MeaningOfLife := 42
end;
{$IFDEF modulino}
begin
    write('Main: The meaning of life is: ');
    writeln(MeaningOfLife())
{$ENDIF}
end.
